src/components/cursor.js

```diff
@@ -196,14 +213,11 @@ module.exports.Component = registerComponent('cursor', {
   },
 
   onMouseMove: (function () {
-    var direction = new THREE.Vector3();
```

Comment:
These variables not meant to store state but used for local calculations and cached to prevent unnecessary instantiations. Why do they need to be public now?

Reply:
The more variables exposed, the larger the API surface area, the greater maintenance burden and increased possibility of misuse.

Reply:
They don't need to be public.  The issue is that they are not only used transiently, but actually passed by reference into the raycaster and used there.

[`this.el.setAttribute('raycaster', rayCasterConfig);`](https://github.com/aframevr/aframe/blob/5da874943f6aa871120f6b865fe9dcd7619e0cea/src/components/cursor.js#L236)

Result is that if you have two separate cursor components on the same scene, they share these variables, and end up scribblng over each others' raycaster parameters.  

It was a bit of a surprise to me that this happened, but it definitely did (took me quite a long time to figure out what was going on, as it was so unexpected - can create a repro if you want to take a look as exactly what happens here).

This was a simple solution that definitely works, because it gives you a unique instance of origin & direction per cursor instance.

I don't know of another way to solve this that doesn't have perf implications (e.g. copying the value of the origin & dircetion vectors to pass them into the raycaster).  Can you suggest a better way to solve this?

Reply:
Would giving them names that indicate they are internal, e.g. `this._orgin` and `this._direction` help?

Or can we make them private?  this.#origin, this.#direction?  (apparently this is new in ES2019, I only just found out about this option)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields